pykrige/tests/test_api.py

```diff
@@ -46,3 +50,43 @@ def test_krige():
                 assert estimator.best_score_ > -3.0
         if hasattr(estimator, 'cv_results_'):
             assert estimator.cv_results_['mean_train_score'] > 0
+
+
+@pytest.mark.skipif(not SKLEARN_INSTALLED,
+                    reason="scikit-learn not installed")
+def test_gridsearch_cv_variogram_parameters():
+    # First dictionary with a dictionary in the place of the variogram_parameters
+    param_dict3d_1 = {"method": ["ordinary3d"], "variogram_model": ["linear"],
+                 "variogram_parameters": [{'slope': 1.0, 'nugget': 1.0},
+                                          {'slope': 2.0, 'nugget': 1.0}]
+               }
+    # First dictionary with a list of lists in the place of the variogram_parameters
+    param_dict3d_2 = {"method": ["ordinary3d"], "variogram_model": ["linear"],
+                 "variogram_parameters": [[1.0,1.0],
+                                          [2.0,1.0]]
+               }
```

Comment:
This is pretty extensive, but I'm not convinced we really do want to compare these two cases. In the end comparing them would be testing that "variogram_parameters" works with a list and a dict (assuming `GridSearchCV` work as expected) and that it produces the same results, which is another check.

So maybe just keep `param_dict3d_1` and check that we have the right size e.g. of `estimator1.cv_results_['param_variogram_parameters'`]` (I think, or some other key there).
And also check that the `best_param_` is the one expected..

That would simplify this test a bit.